libgalois/include/katana/NUMAArray.h

```diff
@@ -130,6 +130,11 @@ class NUMAArray {
   void set(difference_type x, const_reference v) { data_[x] = v; }
   size_type size() const { return size_; }
   bool empty() const { return size() == 0; }
+  // calling front() or back() on an empty array is unsafe
+  reference front() { return data_[0]; }
+  const_reference front() const { return data_[0]; }
+  reference back() { return data_[size_ - 1]; }
```

Comment:
please use `size()` instead of `size_` field. How about adding a debug assert of `!empty()` as that will assist those debugging weird issues.

Reply:
+1 for a debug assert. Seems better than nothing.